client/src/components/SignUp.jsx

```diff
@@ -64,104 +76,117 @@ const SignUp = () => {
       case "password":
         setPassword(value);
         break;
-      case "password2":
-        setPassword2(value);
+      case "confirmedPass":
+        setConfirmedPass(value);
         break;
       case "error":
         setError(value);
         break;
     }
   };
 
-  const submit = (event) => {
+  const submit = async (event) => {
     event.preventDefault();
 
-    if (password.length < 6) {
-      setError("Password needs to be at least 6 characters long");
-    } else if (password === password2) {
-      console.log(userName, email, password, password2);
-      setError("");
+    if (password.length < 7) {
+      setError("Password needs to be at least 7 characters long");
+    } else if (password === confirmedPass) {
+      try {
+        const { data } = await axios.post("/api/users/signup", {
+          name: userName,
+          email: email,
+          password: password,
+        });
+
+        localStorage.token = data.token;
+
+        if (localStorage.token) {
+          setNextPage(true);
+        }
```

Comment:
I feel like removing and adding to localStorage should be in a separate file that contains some util functionality related to it - this way if we want to change the way we store auth, it's easier